File: app/src/main/java/com/whiteboardapp/common/ModelArgs.java
package com.whiteboardapp.common;

import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;

public class ModelArgs {
    public float conf;
    public float iou;
    public boolean agnosticNMS;
    public int maxDet;
    public int nc;
    //ret sikker på den ikke er nødvendig
    public String classes;

    private final String FILE_NAME = "args.txt";


    //Skal omskrives til at læse en fil
    public ModelArgs() {
        conf = 0.25f;
        iou = 0.7f;
        agnosticNMS = false;
        maxDet = 300;
        nc = 1;
        classes = null;
        LoadModelArgsFromFile();
    }

    public void LoadModelArgsFromFile(){
        BufferedReader reader;

        try {
            reader = new BufferedReader(new FileReader(FILE_NAME));
            String line = reader.readLine();

            while(line != null){
                AssignValues(line);
                line = reader.readLine();
            }
        } catch (IOException e){
            e.printStackTrace();
        }
    }

    private void AssignValues(String line){
        String[] splitLine = line.split(":");
        switch(splitLine[0]){
            case "conf":
                    conf = Float.parseFloat(splitLine[1]);
                break;
            case "iou":
                iou = Float.parseFloat(splitLine[1]);
                break;
            case "agnost_nms":
                agnosticNMS = Boolean.valueOf(splitLine[1]);
                break;
            case "max_det":
                maxDet = Integer.parseInt(splitLine[1]);
                break;
            case "nc":
                nc = Integer.parseInt(splitLine[1]);
                break;
            case "classes":
                classes = null;
                break;
            default:
                System.out.println("How did we even get here? ModelArgs broke");

        }
    }
}

File: Model Analysis/ModelArgs.py
class ModelArgs:
    def __init__(self):
        self.Test()

    def Test(self):
        with open("args.txt", "r") as file:
            lines = file.readlines()
            for line in lines:
                key, value = line.strip().split(":")
                
                if key == "conf":
                    if value == "null":
                        self.conf = 0.25
                    else:
                        self.conf = float(value)
                elif key == "iou":
                    self.iou = float(value)
                elif key == "agnostic_nms":
                    self.agnostic_nms = bool(value)
                elif key == "max_det":
                    self.max_det = int(value)
                elif key == "nc":
                    self.nc = int(value)
                elif key == "classes":
                    if value == "none":
                        self.classes = None
                    else:
                        self.classes = value
                else:
                    print("Error! Unknown key detected in args.txt!")